python/cuml/comm/serialize.py

```diff
@@ -25,11 +27,37 @@
 serializable_classes = (cuml.common.CumlArray,)
 
 
+def serialize_mat_descriptor(m):
+    return cp.cusparse.MatDescriptor.create, ()
+
+
 try:
     from distributed.protocol import dask_deserialize, dask_serialize
     from distributed.protocol.cuda import cuda_deserialize, cuda_serialize
     from distributed.utils import log_errors
 
+    from distributed.protocol import register_generic
+
+    from cuml.naive_bayes.naive_bayes import MultinomialNB
+
+    try:
+
+        # cuda serializer requires Dask Distributed 2.12.0+
+        register_generic(MultinomialNB, 'cuda',
+                         cuda_serialize, cuda_deserialize)
+
+        register_generic(cuml.Base, 'cuda',
+                         cuda_serialize, cuda_deserialize)
+
+    except TypeError:
+        pass
+
+    # Registers 'dask' serialization by default
+    register_generic(MultinomialNB)
+    register_generic(cuml.Base)
```

Comment:
Since we are jumping to Dask + Distributed 2.12.0, this can now be simplified.

```suggestion
    register_generic(MultinomialNB, 'cuda',
                     cuda_serialize, cuda_deserialize)
    register_generic(MultinomialNB, 'dask',
                     dask_serialize, dask_deserialize)

    register_generic(cuml.Base, 'cuda',
                     cuda_serialize, cuda_deserialize)
    register_generic(cuml.Base, 'dask',
                     dask_serialize, dask_deserialize)
```

Reply:
Does `register_generic` still register 'dask' serializer when the serializers aren't provided? If that's the case, can't we simplify this even more?

Reply:
Yep, sure.

Though the question came up before about whether we want `register_generic(cls)` to have different behavior (like also including `"cuda"`). So this would avoid making any assumptions about what that does.